presto-main/src/test/java/com/facebook/presto/operator/scalar/TestRegexpFunctions.java

```diff
@@ -141,6 +141,69 @@ public void testRegexpReplace()
+        assertFunction("REGEXP_REPLACE('abc abc', 'abc', x -> 'm')", createUnboundedVarcharType(), "m m"); // Empty block passed to lambda
+        assertFunction("REGEXP_REPLACE('abc', '', x -> 'OK')", createUnboundedVarcharType(), "OKaOKbOKcOK");  // Empty block passed to lambda
+        assertFunction("REGEXP_REPLACE('abc', '()', x -> x[1])", createUnboundedVarcharType(), "abc"); // Passed a block containing multiple empty capturing groups to lambda
+        assertFunction("REGEXP_REPLACE('abc', '()', x -> 'OK')", createUnboundedVarcharType(), "OKaOKbOKcOK"); // Passed a block containing multiple empty capturing groups to lambda
+        assertFunction("REGEXP_REPLACE('new', '(\\w*)', x -> upper(x[1]))", createUnboundedVarcharType(), "NEW");  // Two matches: ["new"] and [""]
+        assertFunction("REGEXP_REPLACE('new', '(\\w*)', x -> x[1] || upper(x[1]))", createUnboundedVarcharType(), "newNEW");  // Two matches: ["new"] and [""]
+        assertFunction("REGEXP_REPLACE('new', '(\\w*)', x -> cast(length(x[1]) AS varchar))", createUnboundedVarcharType(), "30");  // Two matches: ["new"] and [""]
+        assertFunction("REGEXP_REPLACE('new york', '(\\w*)', x -> '<' || x[1] || '>')", createUnboundedVarcharType(), "<new><> <york><>");  // Four matches: ["new"], [""], ["york"], [""]
+
+        // Matches that contains null capturing groups
+        assertFunction("REGEXP_REPLACE('aaa', '(b)?', x -> x[1] )", createUnboundedVarcharType(), null);
+        assertFunction("REGEXP_REPLACE('abde', 'ab(c)?de', x -> x[1])", createUnboundedVarcharType(), null);  // Matched once with two matching groups[(0,3),null].
+        assertFunction("REGEXP_REPLACE('abde', 'ab(c)?de', x -> 'OK')", createUnboundedVarcharType(), "OK");  // Matched once with two matching groups[(0,3),null]. Passed null to lambda and returns OK, so whole string replace with OK
+        assertFunction("REGEXP_REPLACE('abde', 'ab(c)?de', x -> x[1] || 'OK')", createUnboundedVarcharType(), null); // Passed null to lambda and returns null.
+        assertFunction("REGEXP_REPLACE('abde', 'ab(c)?de', x -> 'OK' || x[1])", createUnboundedVarcharType(), null);  // Passed null to lambda and returns null.
+        assertFunction("REGEXP_REPLACE('abde', 'ab(c)?de', x -> case when x[1] is null then 'foo' else 'bar' end)", createUnboundedVarcharType(), "foo");
+        assertFunction("REGEXP_REPLACE('ab', '(a)?(b)?', x -> case when (x[1] is not null) AND (x[2] is not null) then 'foo' else null end)", createUnboundedVarcharType(), null);
+
+        // Matches that contains non-empty and not null capturing groups but lambda returns null
+        assertFunction("REGEXP_REPLACE('aaa', '(a)', x -> cast(null as varchar))", createUnboundedVarcharType(), null);
+        assertFunction("REGEXP_REPLACE('ab', '(a)?(b)?', x -> case when (x[1] is not null) AND (x[2] is null) OR (x[1] is null) AND (x[2] is NOT null) then 'foo' else null end)", createUnboundedVarcharType(), null);
```

Comment:
Nit: capitalize SQL keywords like `CASE`, `WHEN`, `IS NOT NULL`, etc. Same for the other tests.